packages/common/src/store/solana/sagas.ts

```diff
@@ -22,6 +23,13 @@ function* watchForFeePayer() {
   }
 }
 
+export function* waitForFeePayer() {
```

Comment:
there is probably a bunch of other places that could use this

Reply:
this pattern is somewhat error prone b/c of race condition:
1. `setFeePayer` is dispatched
2. someone calls `waitForFeePayer`
3. fee payer gets set
4. take never gets called

worth pointing out, but probably not critical.

the way to do it is probably a `waitForValue` on `getFeePayer`